src/main/java/se/liu/ida/hefquin/engine/queryplan/executable/impl/ops/ExecOpFilter.java

```diff
@@ -27,7 +29,24 @@ public int preferredInputBlockSize() {
 	public void process( final IntermediateResultBlock input,
 	                     final IntermediateResultElementSink sink,
 	                     final ExecutionContext execCxt ) throws ExecOpExecutionException {
-		// TODO Auto-generated method stub
+		
+		
+		// For every solution mapping in the input...
+		for(SolutionMapping solution : input.getSolutionMappings()) {
+			//Check whether it satisfies the filter expression
+			
+			try {
+				final NodeValue evaluationResult = ExprUtils.eval(filterExpression, solution.asJenaBinding());
+				if(evaluationResult == NodeValue.TRUE) {
+					sink.send(solution);
+				} else if (evaluationResult != NodeValue.FALSE ) {
+					throw new ExecOpExecutionException("The result of the eval is neither TRUE nor FALSE!", null);
+				}
+			} catch (VariableNotBoundException e) {
+				throw new ExecOpExecutionException("A variable name in the binding does not match the variable names in the filter.", null);
```

Comment:
You should not throw an exception here. The `VariableNotBoundException` that triggers this catch block indicates that evaluating the filter expression based on the current solution mapping results in an error, which means that the current solution mapping does not satisfy the filter condition and, thus, is not in the output of this operator (i.e., should not be sent to the sink). Hence, instead of throwing an exception here, simply do not do anything in this catch block, which will then cause the program flow to move on to the next solution mapping from the input.

Having said that, instead of leaving this catch block completely empty, it may be helpful to include a comment about why the catch block does not do anything. So, here is my change suggestion for this source code line:
```suggestion
				// The current solution mapping does not satisfy the filter condition (because
				// evaluating the filter expression based on this solution mapping resulted in
				// this error). Therefore this solution mapping is not in the output of this operator
				// and, thus, must not be sent to the sink. Hence, we do not have to do anything
				// here in this catch block.
```